test/matchers/log_format.go

```diff
@@ -0,0 +1,139 @@
+	expected interface{}
+}
+
+func FitLogFormatTemplate(expected interface{}) types.GomegaMatcher {
+	return &LogMatcher{
+		expected: expected,
+	}
+}
+
+func (m *LogMatcher) Match(actual interface{}) (success bool, err error) {
+	template, ok := m.expected.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+	log, ok := actual.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+
+	return CompareLog(template,log)
+}
+
+func (m *LogMatcher) FailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto fit \n\t%#v", actual, m.expected)
+}
+
+func (m *LogMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto not fit \n\t%#v", actual, m.expected)
+}
+
+func isNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		return reflect.ValueOf(i).IsNil()
+	}
+	return false
+}
+
+func DeepFields(iface interface{}, namePrefix string) ([]reflect.Value, []string) {
+	fields := make([]reflect.Value, 0)
+	names :=  make([]string,0)
+
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+
+	for i := 0; i < ift.NumField(); i++ {
+		v := ifv.Field(i)
+		n := namePrefix + ifv.Type().Field(i).Name
+
+		switch v.Kind() {
+		case reflect.Struct:
+			moreFields, moreNames := DeepFields(v.Interface(), n + "_")
+			fields = append(fields, moreFields...)
+			names = append(names, moreNames...)
+		case reflect.Ptr:
+			if !isNil(v.Interface()) {
+				elm := v.Elem()
+				moreFields, moreNames := DeepFields(elm.Interface(), n + "_")
+				fields = append(fields, moreFields...)
+				names = append(names, moreNames...)
+			}
+		default:
+			fields = append(fields, v)
+			names = append(names, n)
+		}
+	}
+
+	return fields, names
+}
+
+func compareLogLogic (name string, templateValue interface{}, value interface{}) bool {
+	if templateValue == value { // Same value is ok
+		logger.V(3).Info("CompareLogLogic: Same value for", "name", name, "value", value)
+		return true
+	}
+	if templateValue == "*" && !isNil(value) { // Any value, not Nil is ok if template value is "*"
+		logger.V(3).Info("CompareLogLogic: Any value for * ", "name", name, "value", value)
+		return true
+	}
+	if strings.HasPrefix(templateValue.(string), "Pre-") { // Any prefix value is ok if template value starts with "Pre-"
+		if strings.HasPrefix(value.(string)[4:], templateValue.(string)[4:])  {
+			logger.V(3).Info("CompareLogLogic: Fit prefix ", "name", name, "value", value)
+			return true
+		}
+	}
+
+	logger.V(3).Info("CompareLogLogic: Mismatch", "name", "templateValue", templateValue, "value", value)
+	return false
+}
+
+func CompareLog(template Log, log Log) (bool, error) {
+	templateFieldValues, templateFieldNames := DeepFields(template,"")
+	logFieldValues, logFieldNames := DeepFields(log,"")
+	for i := 0; i < len(logFieldValues); i++ {
```

Comment:
why not range where you can get the number and entry during interation?

Reply:
fixed